src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -406,7 +412,67 @@ internal T WithProofOfPosession(HttpRequestMessage httpRequestMessage, IPoPCrypt
             CommonParameters.AddApiTelemetryFeature(ApiTelemetryFeature.WithPoPScheme);
             CommonParameters.AuthenticationScheme = new PoPAuthenticationScheme(httpRequestMessage, popCryptoProvider);
 
-            return this as T;
+            return (T)this;
+        }
+
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item> An Authentication header is automatically added to the request</item>
+        /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
+        /// <item>MSAL creates, reads and stores a key in memory that will be cycled every 8 hours.</item>
+        /// </list>
+        /// </remarks>
+        public T WithProofOfPosession()
+        {
+            CommonParameters.UsingProofOfPossesion = true;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// Specifies the HTTP method of the HTTP request to the protected resource which requires a PoP token. 
+        /// The PoP token will be cryptographically bound to the request.
+        /// </summary>
+        /// <param name="httpMethod">Http Method for proof of possesion request.</param>
+        /// <returns></returns>
+        public T WithProofOfPosessionMethod(HttpMethod httpMethod)
+        {
+            CommonParameters.PopMethod = httpMethod;
+            return (T)this;
+        }
+
+        /// <summary>
+        /// Specifies the URL of the HTTP request to the protected resource which requires a PoP token. 
+        /// The PoP token will be cryptographically bound to the request.
+        /// </summary>
+        /// <param name="uri">Protected resource URL.</param>
+        /// <returns></returns>
+        public T WithProofOfPosessionUri(Uri uri)
```

Comment:
I don't like this strategy of piling on options on the AcquireToken* method because it "polutes" . There will be at least 4 params on SHR -> 4 methods on each AcqurieToken* entry point. 

Instead, you can add a simple config object, similar to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/530d9f08b2c5f59edc2ad1dd82ca0729114e271e/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenInteractiveParameterBuilder.cs#L125

So we could have a single method:

`WithPoP(PoPOptions options = null)`

Reply:
The main question here is - what should the default be here? I would argue for a secure default, where POP takes into account the method (GET / POST) and the url ...